### PR TITLE
feat: add hint for new noarch: python syntax

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry
* [x] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

This PR adds hints for the new `noarch: python` syntax. They are not lints since this syntax is not required per the CFEP.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/core this one is ready for review!

File: conda_smithy/linter/hints.py [hunk 1]
```diff
@@ -232,3 +232,29 @@ def hint_pip_no_build_backend(host_or_build_section, package_name, hints):
                 "If your recipe has built with only `pip` in the `host` section in the past, you likely should "
                 "add `setuptools` to the `host` section of your recipe."
             )
+
+
+def hint_noarch_python_use_python_min(
+    host_reqs, run_reqs, test_reqs, outputs_section, noarch_value, hints
+):
+    if noarch_value == "python" and not outputs_section:
```
Comment (jaimergp) [inline conda_smithy/linter/hints.py:240]: We don't hint multi output recipes? 🤔

File: conda_smithy/linter/hints.py [hunk 2]
```diff
@@ -232,3 +232,29 @@ def hint_pip_no_build_backend(host_or_build_section, package_name, hints):
...
+                if (
+                    req.strip().split()[0] == "python"
+                    and req != "python"
+                    and syntax in req
+                ):
+                    break
+            else:
+                hints.append(
+                    f"noarch: python recipes should almost always follow the syntax in "
+                    f"[CFEP-25](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python). "
```
Comment (jaimergp) [inline conda_smithy/linter/hints.py:256]: Is this supposed to be a link to CFEP-25?
```suggestion
                    f"[`noarch: python`](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) recipes should almost always follow the syntax in "
                    f"[CFEP-25](https://github.com/conda-forge/cfep/blob/main/cfep-25.md). "
```

File: conda_smithy/linter/hints.py (hunk 1 above)
Comment (beckermr) [reply conda_smithy/linter/hints.py:240]: Not currently no. IDK why.

File: conda_smithy/linter/hints.py (hunk 2 above)
Comment (beckermr) [reply conda_smithy/linter/hints.py:256]: It was supposed to be a link to the docs. I can change the link text.

Comment (beckermr) [reply conda_smithy/linter/hints.py:256]:
```suggestion
                    f"noarch: python recipes should almost always follow the syntax in "
                    f"our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python). "
```

File: conda_smithy/linter/hints.py (hunk 1 above)
Comment (jaimergp) [reply conda_smithy/linter/hints.py:240]: @isuruf added similar code in https://github.com/conda-forge/conda-smithy/commit/b7514aa03125d7dcc13de6da1206577c34ac69bf

There's no PR attached to that blame so I'm not sure where it came from or whether there was some discussion. We should probably make it compatible with multi output recipes but I won't block if you consider it's out of scope.

Comment (beckermr) [reply conda_smithy/linter/hints.py:240]: I think it is out of scope for this PR given that it is already in the code base.

**Review by jaimergp — APPROVED** [review]

**Pull request merged**